firmware/init/sensor/init_tps.cpp

```diff
@@ -181,19 +182,32 @@ void initTps() {
 
 	if (!engineConfiguration->consumeObdSensors) {
 		bool isFordTps = engineConfiguration->useFordRedundantTps;
+		bool isFordPps = engineConfiguration->useFordRedundantPps;
 
-		tps1.init(isFordTps, &fordTps1,
+		float tpsSecondaryMaximum = engineConfiguration->tpsSecondaryMaximum;
+		if (tpsSecondaryMaximum < 10 || tpsSecondaryMaximum > 90) {
+			// most ford TPS seem to be ~52.6%, default to that if not set
+			tpsSecondaryMaximum = 52.6f;
+		}
+
+		tps1.init(isFordTps, &fordTps1, tpsSecondaryMaximum,
 			{ engineConfiguration->tps1_1AdcChannel, (float)engineConfiguration->tpsMin, (float)engineConfiguration->tpsMax, min, max },
 			{ engineConfiguration->tps1_2AdcChannel, (float)engineConfiguration->tps1SecondaryMin, (float)engineConfiguration->tps1SecondaryMax, min, max }
 		);
 
-		tps2.init(isFordTps, &fordTps2,
+		tps2.init(isFordTps, &fordTps2, tpsSecondaryMaximum,
 			{ engineConfiguration->tps2_1AdcChannel, (float)engineConfiguration->tps2Min, (float)engineConfiguration->tps2Max, min, max },
 			{ engineConfiguration->tps2_2AdcChannel, (float)engineConfiguration->tps2SecondaryMin, (float)engineConfiguration->tps2SecondaryMax, min, max }
 		);
 
+		float ppsSecondaryMaximum = engineConfiguration->ppsSecondaryMaximum;
+		if (ppsSecondaryMaximum < 10 || ppsSecondaryMaximum > 90) {
```

Comment:
This range-resetting (and limiting) behavior seems unfriendly (and unnecessarily restrictive).
1) it'd be better to clamp the value than reset it without notice
2) it'd be better to just accept the given value (with the possible exception of 0, and _maybe_ 100, defaulting to something "sane").

I'd argue the same about the TPS application, but not knowing the details of the original concern about Ford TPS I can't argue against retaining it (though the unnoticed reset is still unfriendly).

Reply:
The concern about being too lenient is that you can set it to 0%, which will defeat the redundancy on your throttle completely. I've changed it to just not allow small values, but allow 100 (since that's actually safe)